lib/archethic/networking/ip_lookup/nat_discovery/miniupnp.ex

```diff
@@ -0,0 +1,65 @@
+defmodule Archethic.Networking.IPLookup.NATDiscovery.MiniUPNP do
+  @moduledoc false
+
+  require Logger
+
+  @upnpc Application.app_dir(:archethic, "priv/c_dist/upnpc")
+
+  @spec get_node_ip() :: {:ok, :inet.ip_address()} | {:error, any()}
+  def get_node_ip do
+    case System.cmd(@upnpc, ["-s"]) do
+      {output, 0} ->
+        [[_, ip]] = Regex.scan(~r/ExternalIPAddress = ([0-9.]*)/, output, capture: :all)
+
+        ip
+        |> to_charlist()
+        |> :inet.parse_address()
+
+      {_, status} ->
+        {:error, status}
+    end
+  end
+
+  @spec open_port(non_neg_integer()) :: :ok | :error
+  def open_port(port) do
+    # {:ok, [{local_ip, _, _} | _]} = :inet.getif()
+
+    local_ip =
+      case System.cmd(@upnpc, ["-s"]) do
+        {output, 0} ->
+          [[_, ip]] = Regex.scan(~r/Local LAN ip address : ([0-9.]*)/, output, capture: :all)
+
+          ip
+          |> to_charlist()
+          |> :inet.parse_address()
+          |> elem(1)
+
+        {_, status} ->
+          {:error, status}
+      end
+
+    opts = [
+      # Add redirection
```

Comment:
`-e "ArchethicPort"`
 Is it a good thing to have this?

Reply:
why not

Reply:
Do I do a that by committing on add_miniupnp?

Reply:
I will do it ;)